### ISSUE TITLE
tracer.NewNoopTracer support

### ISSUE THREAD

**Issue by jcchavezs** [openzipkin/zipkin-go#65]
Currently when writing a test that has a tracer as dependency, there is no easy way to create a `noop` tracer.

Ping @basvanbeek @adriancole

**Comment by basvanbeek** [comment]
NewTracer(nil) gives you a noop tracer. I'd prefer improved documentation on NewTracer() instead of adding a convenience function.

**Comment by jcchavezs** [comment]
Problem with `NewTracer(nil)` is that it returns an error, making it very inconvenient:

```
noopTracer, err := NewTracer(nil)
// do something with the error
```

In addition, `NewTracer(nil)` does not give me an actual noop tracer but a tracer that does not do reporting instead.

**Comment by basvanbeek** [comment]
WYTBW?  

NewTracer(nil) returns a noop implementation, see:
https://github.com/openzipkin/zipkin-go/blob/master/tracer.go#L43

You can start with:
```go
noopTracer, _ := NewTracer(nil)
```

**Comment by jcchavezs** [comment]
🎖 I usually don't like ignore such errors but in this case I think it makes sense. Won't close this issue to keep a reminder to improve docs.

**Comment by basvanbeek** [comment]
agreed... typically one will always check for the error when calling NewTracer... only time you'd hard code a `NewTracer(nil)` is in tests.
